### DISCUSSION TITLE
Is it possible to load an anvil world packaged into a JAR?

### DISCUSSION THREAD

**Discussion by offby0point5** [Minestom/Minestom#1191] [Help] [answered]
I want to have a server loading its world directly from the jar. Copying the world to disk and using AnvilLoader would be possible, but this would use unnecessary disk space as the world should be read only anyway.
How can I load it directly from the JAR?

**Comment by VegetalDev** [comment]
This should be (theoretically) possible, but you'd have to basically rewrite AnvilLoader to support loading region "files" and other necessary data from arbitrary InputStreams, not only those created from Paths. Currently, the use of Path objects is hardcoded in the constructor (`levelPath` and `regionPath` are both resolved there) so extending isn't really a good option. 

I think it would be useful to have a more flexible option that still uses the Anvil format but doesn't rely on files. Have you considered making this into an issue?

**Comment by offby0point5** [comment, marked as answer]
Anvil doesn't work with this, but there are two alternatives for Anvil:
- Slime, which, as far as I know, needs a setup of spigot and the plugin for slime to be able to convert from Anvil. The Minestom library for Slime has no write support until now
- TNT. which is only available for Minestom, but has read and write support (if not packaged in a JAR) and is in my opinion easier to convert to form Anvil
